bin/i18n/plugins/blockly.js

```diff
@@ -0,0 +1,79 @@
+let redact;
+
+const BLOCKLY = 'blockly';
+
+module.exports = function blockly() {
+  if (this.Parser) {
+    const Parser = this.Parser;
+    redact = Parser.prototype.options.redact;
+
+    Parser.prototype.restorationMethods[BLOCKLY] = function(add, node) {
+      return add({
+        type: 'rawtext',
+        value: node.original
+      });
+    };
+
+    ['inline', 'block'].forEach(type => {
+      Parser.prototype[`${type}Tokenizers`][BLOCKLY] = tokenizeBlockly;
+      const methods = Parser.prototype[`${type}Methods`];
+      methods.splice(methods.indexOf('html'), 0, BLOCKLY);
+    });
+  }
+};
+
+function tokenizeBlockly(eat, value, silent) {
+  // This plugin ONLY exists to redact Blockly blocks from translations, not to
+  // render anything.
+  if (!redact) {
+    return false;
+  }
+
+  if (!value.startsWith('<xml>')) {
+    return false;
+  }
+
+  // Search indiscriminately for next closing tag. Note that this precludes
+  // situations like:
+  //
+  // <xml>
+  //   <block>
+  //     <xml>
+  //       <block/>
+  //     </xml>
+  //   </block>
+  // </xml>
+  //
+  // Because we would match the inner closing tag with the outer opening tag.
+  // However, that's not valid XML anyway, so I don't think we care.
+  const closingTag = '</xml>';
+  const closingIndex = value.indexOf(closingTag);
+
+  if (closingIndex === -1) {
+    return false;
+  }
+
+  if (silent) {
+    return true;
+  }
+
+  // Eat to end of closing tag
+  const original = value.substring(0, closingIndex + closingTag.length);
+  return eat(original)({
+    type: 'redaction',
+    redactionType: BLOCKLY,
+    original,
+    children: [
+      {
+        type: 'text',
+        value: 'blockly block'
```

Comment:
So all redactions will just say "blockly block"? Is it worth trying to be more descriptive or would that be too confusing?

Reply:
Yep!

Note that we won't use whatever value translators put in here anywhere, so I debated adding any text at all. I think "blockly block" is sufficiently descriptive, but we could easily change it in the future if we decide we want something more. I was also thinking of just using the string "blockly"

For example, consider https://studio.code.org/s/coursea-2018/stage/4/puzzle/2. The text:

`To get the bird to the pig, snap the <xml><block type="maze_moveEast"/></xml> block to the bottom of the <xml><block type="when_run"/></xml> block, then press "▶ Run"!`

will be redacted to:

`To get the bird to the pig, snap the [blockly block][0] block to the bottom of the [blockly block][1] block, then press "▶ Run"!`

Reply:
That makes sense. Thanks for the explanation!